### PR TITLE
Handling encryption with matrix connection in the Dockerfile build

### PR DESCRIPTION
**Pull request by dud1337**
The matrix connection's encryption handling doesn't work with the current dev docker build due to missing dependencies. `nio.crypto.ENCRYPTION_ENABLED` returns `False`

To solve this, 3 dependencies are installed:
* `libffi-dev` via `apk`
* `python-olm` via `pip`
* `matrix-nio[e2e]` via `pip`

I've tested this in an encrypted room.

It should be noted that this by default **doesn't have persistence on the generated device key** and it can be a tad annoying if you name the device the same thing: Matrix doesn't see the new key. Using a volume for the `matrix_store` solves this.

Relates to #1676

### CODE REVIEW

**Comment by jacobtomlinson** [comment]
cc @Cadair

File: setup.cfg [hunk 1]
```diff
@@ -82,6 +82,8 @@ console_scripts =
 connector_matrix =
 	bleach>=3.1.5
 	matrix-nio>=0.14.1
+	python-olm
+	matrix-nio[e2e]
```
Comment (Cadair) [inline setup.cfg:86]: I don't think we want this here like this.

There are two things, I think `python-olm` should be a dependency of `matrix-nio[e2e]` anyway, also I think that we don't want to require e2e for installing the matrix connector, I think it needs to be opt-in as it's difficult to install on non-linux platforms at the moment.

What I propose is this:
```suggestion
connector_matrix_e2e =
	matrix-nio[e2e]
```

**Comment by dud1337** [comment]
>which will add a new extra so you would do pip install opsdroid[connector_matrix, connector_matrix_e2e] to get all the e2e deps.

Sounds good. As long as it's in the docker image for my laziness :]

I'll say that in the long run, from the matrix perspective, having e2e as "optional" feels wrong, but I understand re: installation difficulties on other platforms.

**Comment by Cadair** [comment]
> I'll say that in the long run, from the matrix perspective, having e2e as "optional" feels wrong

I agree, and the packaging situation is improving for the olm library and bindings, but basically until it's on PyPI it's impractical for us to install it by default.

**Review by Cadair — APPROVED** [review]

**Comment by Cadair** [comment]
@jacobtomlinson Are the test fails here expected?

**Review by Cadair — CHANGES_REQUESTED** [review]
Well I am going to assume this test fail isn't expected :stuck_out_tongue: 

https://github.com/opsdroid/opsdroid/pull/1690/checks?check_run_id=1645033142#step:5:463

**Comment by Cadair** [comment]
It seems that it's not finding olm when installing python-olm?

**Comment by Cadair** [comment]
I looked into the error and got a little carried away, I hope you don't mind.

I have changed two things:

1. I excluded the new `connector_matrix_e2e` from being included in the `all` extra.
2. I skipped the install step for the docker builds to save a little time.

**Review by Cadair — APPROVED** [review]

**Comment by dud1337** [comment]
Thanks; I was dreading fixing the docker builds on windows anyway.

**Pull request merged**